### PR TITLE
IGAPP-570: City bounding box

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: web/src/CityContentSwitcher.tsx [hunk 1]
```diff
@@ -140,34 +140,63 @@ const CityContentSwitcher = ({ cities, match, location }: PropsType): ReactEleme
 
   const routes: ReactElement[] = []
   if (eventsEnabled) {
-    routes.push(<Route render={render(EVENTS_ROUTE, EventsPage)} path={RoutePatterns[EVENTS_ROUTE]} exact />)
+    routes.push(
+      <Route key={EVENTS_ROUTE} render={render(EVENTS_ROUTE, EventsPage)} path={RoutePatterns[EVENTS_ROUTE]} exact />
```
Comment (klinzo) [inline web/src/CityContentSwitcher.tsx:144]: Only added keys to all routes to fix react error.

**Review by f1sh1918 — APPROVED** [review]

File: api-client/src/Endpoint.ts [hunk 2]
```diff
@@ -83,6 +84,7 @@ class Endpoint<P, T> {
       const fetchedData = this.mapResponse(json, params)
       return new Payload(false, url, fetchedData, null)
     } catch (e) {
+      console.error(e)
```
Comment (Taggotty) [inline api-client/src/Endpoint.ts:87]: Is it best practice to log the error? Just asking :)

File: api-client/src/models/CityModel.ts [hunk 3]
```diff
@@ -32,13 +27,8 @@ class CityModel {
...
     longitude: number | null
-    aliases: Record<
-      string,
-      {
-        longitude: number
-        latitude: number
-      }
-    > | null
+    aliases: Record<string, { longitude: number; latitude: number }> | null
+    boundingBox: { latitude: number; longitude: number }[] | null
```
Comment (Taggotty) [inline api-client/src/models/CityModel.ts:31]: Could you extract the bounding box type here? You seem to reuse `{ latitude: number; longitude: number }` quite often

**Review by Taggotty — APPROVED** [review]

File: api-client/src/models/CityModel.ts (hunk 3 above)
Comment (f1sh1918) [reply api-client/src/models/CityModel.ts:31]: Another question is if we should not use the standard format for a bounding box refer to geoJSON
```js
/**
 * Bounding box
 * https://tools.ietf.org/html/rfc7946#section-5
 */
export type BBox = [number, number, number, number] | [number, number, number, number, number, number];
```

File: api-client/src/Endpoint.ts (hunk 2 above)
Comment (klinzo) [reply api-client/src/Endpoint.ts:87]: Not sure but I definitely find it useful. It is stripped in production anyway.

File: api-client/src/models/CityModel.ts (hunk 3 above)
Comment (klinzo) [reply api-client/src/models/CityModel.ts]: @f1sh1918 sorry I don't have time to adjust this right now, but feel free to change this :)

File: api-client/src/endpoints/createCitiesEndpoint.ts [hunk 4]
```diff
@@ -36,7 +37,10 @@ export default (baseUrl: string): Endpoint<void, Array<CityModel>> =>
               prefix: city.prefix,
               longitude: city.longitude,
               latitude: city.latitude,
-              aliases: city.aliases
+              aliases: city.aliases,
+              boundingBox: Array.isArray(city.bounding_box)
+                ? (city.bounding_box.reduce((acc, val) => acc.concat(val), []) as BBox)
```
Comment (f1sh1918) [inline api-client/src/endpoints/createCitiesEndpoint.ts:42]: Maybe someone can take a short look how to properly type this. Couldn't find a proper solution instantly

File: api-client/src/endpoints/createCitiesEndpoint.ts [hunk 5]
```diff
@@ -36,7 +37,10 @@ export default (baseUrl: string): Endpoint<void, Array<CityModel>> =>
               prefix: city.prefix,
               longitude: city.longitude,
               latitude: city.latitude,
-              aliases: city.aliases
+              aliases: city.aliases,
+              boundingBox: Array.isArray(city.bounding_box)
+                ? (city.bounding_box.reduce((minPoint, maxPoint) => [...minPoint, ...maxPoint]) as BBox)
```
Comment (f1sh1918) [inline api-client/src/endpoints/createCitiesEndpoint.ts:42]: if someone finds a good way to type it, feel free :)

**Review by maxammann — APPROVED** [review]

**Review by Taggotty — APPROVED** [review]

**Comment by klinzo** [comment]
@f1sh1918 if you want you can merge this. Added one minor optional suggestion :)

File: api-client/src/endpoints/createCitiesEndpoint.ts (hunk 5 above)
Comment (f1sh1918) [reply api-client/src/endpoints/createCitiesEndpoint.ts:42]: I think you mean [1] in the second case

Comment (klinzo) [reply api-client/src/endpoints/createCitiesEndpoint.ts:42]: I do :grin:

Comment (klinzo) [reply api-client/src/endpoints/createCitiesEndpoint.ts:42]: If you feel like the current solution is easier/more intuitive just keep it, just stumbled upon it. I also think you are working too late ^^

Comment (f1sh1918) [reply api-client/src/endpoints/createCitiesEndpoint.ts:42]: no i thought omg why was your solution not my first idea...

**Pull request merged**